### PR TITLE
feat(sbom_issue): Support community issue #7247

### PR DESCRIPTION
**Pull request by russbelln**
**Description**
Support issue #7247, to promote the practice of SBOM in the tool, being able to associate components that do not present vulnerabilities.

- Added the **Component** model to the system.  
- Implemented database migrations to support the new model.  
- Updated the importer and re-importer functionalities to create or associate **Component** objects with findings.  
- Enabled full **CRUD** operations for the **Component** model.  
- Adjust Component view and Product Component view to manage filtering and the new model.

### CODE REVIEW

File: dojo/db_migrations/0225_component.py
Comment (Gaviria9601) [inline dojo/db_migrations/0225_component.py:1]: delete this migration and create a new single file to do the migration.

File: dojo/db_migrations/0226_finding_component_alter_component_name.py
Comment (Gaviria9601) [inline dojo/db_migrations/0226_finding_component_alter_component_name.py:1]: delete this migration and create a new single file to do the migration.

File: dojo/db_migrations/0228_merge_20241108_1450.py
Comment (Gaviria9601) [inline dojo/db_migrations/0228_merge_20241108_1450.py:1]: delete this migration and create a new single file to do the migration.

File: dojo/db_migrations/0229_alter_component_unique_together.py
Comment (Gaviria9601) [inline dojo/db_migrations/0229_alter_component_unique_together.py:1]: delete this migration and create a new single file to do the migration.

**Review by Gaviria9601 — COMMENTED** [review]
Please, review comments.

File: dojo/db_migrations/0225_component.py
Comment (russbelln) [reply dojo/db_migrations/0225_component.py:1]: File was removed.

File: dojo/db_migrations/0226_finding_component_alter_component_name.py
Comment (russbelln) [reply dojo/db_migrations/0226_finding_component_alter_component_name.py:1]: File was removed.

File: dojo/db_migrations/0228_merge_20241108_1450.py
Comment (russbelln) [reply dojo/db_migrations/0228_merge_20241108_1450.py:1]: File was removed.

File: dojo/db_migrations/0229_alter_component_unique_together.py
Comment (russbelln) [reply dojo/db_migrations/0229_alter_component_unique_together.py:1]: File was removed and add just one file.

**Review by Gaviria9601 — APPROVED** [review]
💯 Thanks

**Pull request merged**